faust/app/base.py

```diff
@@ -968,14 +963,27 @@ def _on_rebalance_when_stopped(self) -> None:
         """
         if self.should_stop:
             return self._on_rebalance_when_stopped()
-        session_timeout = self.conf.broker_session_timeout
+        # shave some time off so we timeout before the broker
+        # (Kafka does not send error, it just logs)
+        session_timeout = self.conf.broker_session_timeout * 0.95
         self.unassigned = not assigned
+
+        revoked: Set[TP]
+        newly_assigned: Set[TP]
+        if self._assignment is not None:
+            revoked = self._assignment - assigned
+            newly_assigned = assigned - self._assignment
+        else:
+            revoked = set()
+            newly_assigned = assigned  # noqa XXX unused
+        self._assignment = assigned
```

Comment:
Should we be setting `self._assignment` here? We can look up the current assignment at any point by looking at `self.consumer.assigment()`. We should maybe use that or create a `property` out of it. Setting this variable, which should ideally track the value there, seems a bit fragile to me.

Reply:
This is to remember what the previous assignment was, it's used to get the list of actual revoked and newly_assigned partitions!